### ISSUE TITLE
Shellcraft is effectively pre-processed

### ISSUE THREAD

**Issue by zachriggle** [Gallopsled/pwntools#452]
The current implementation of shellcraft relies heavily on `${mov}` macros which destroy the semantic value of the values being shuttled around.

In particular, constants are converted from their e.g. `SYS_execve` to `11`.

This removes the distinction between `shellcraft -f asm` and `shellcraft -f p`.

#### 2.1.3

```
shellcraft i386.linux.sh
...
    ;; Call execve("/bin//sh", 0, 0)
    mov al, SYS_execve
    mov ebx, esp
    int 0x80
```

#### Current `master`

```
shellcraft i386.mov eax SYS_execve -fa
...
    /* call execve('esp', 0, 0) */
    mov ebx, esp
    xor ecx, ecx
    push 0xb
    pop eax
    cdq /* Set edx to 0, eax is known to be positive */
    int 0x80
```

**Comment by zachriggle** [comment]
FWIW, this appears fixed in Binjitsu.

```
$ shellcraft i386.mov eax SYS_execve -fa
    push (SYS_execve) /* 0xb */
    pop eax
```

**Comment by Idolf** [comment]
I agree, this seems to have been fixed in pwntools as weel after #636.
